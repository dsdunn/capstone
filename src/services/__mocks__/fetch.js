export const getAllCollections = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'string',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getUserCollections = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'string',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getCollectionsByCategory = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'vinyl',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getCollection = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'vinyl',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getUserInfo = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666
    }])
  })

export const getSearchResults = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      collections:[]
    }])
  })

export const postCollection = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'vinyl',
      title: 'string',
      description: 'string',
      image: 'string'
    }])
  })