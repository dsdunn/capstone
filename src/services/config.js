export default {
  apiKey: '66642069',
  url: 'https://collecshare.herokuapp.com/api/v1'
}