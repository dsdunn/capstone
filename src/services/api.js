import config from './config';

export const apiFetch = (path, method = 'GET', payload = null) => {
  let body = payload ? JSON.stringify(payload) : null
  return fetch(`${config.url}${path}`, {
    method,
    headers: {
     'Content-Type': 'application/json',
     'Accept': 'application/json'
    },
    body
  })
}

export const apiFormDataFetch = (path, method = 'PUT', payload = null) => {
  console.log(path, method, payload);
  let body = payload; 
  return fetch(`${config.url}${path}`, {
    method,
    body
  })
}
