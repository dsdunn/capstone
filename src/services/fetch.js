import { apiFetch, apiFormDataFetch } from './api'

export const postUserInfo = user => {
  return apiFetch('/users', 'POST', user)
}

export const getUserInfo = uid => {
  return apiFetch(`/users/${uid}`, 'GET')
  .then(response => response.json())
}

export const putUserInfo = (userInfo, uid) => {
  return apiFormDataFetch(`/users/${uid}`, 'PUT', userInfo)
  .then(response => console.log(response))
  .then(response => response.json())
}

export const postCollection = collection => {
  return apiFetch(`/collections`, `POST`, collection)
  .then(response => response.json())
}

export const getAllCollections = () => {
  return apiFetch('/collections')
  .then(response => response.json())
}

export const getUserCollections = uid => {
  return apiFetch(`/collections?uid=${uid}`)
  .then(response => response.json())
}