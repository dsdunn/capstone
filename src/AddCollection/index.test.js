import React from 'react';
import { AddCollection } from './';
import { shallow } from 'enzyme';
import { postCollection } from '../services/fetch';

jest.mock('../services/fetch.js');

describe('AddCollection', () => {
  let wrapper;
  let mockHandleChange = jest.fn();
  let mockEvent = {
    target: {name: 'title', value: 'sweet'},
    preventDefault: jest.fn()
  }

  beforeEach(() => {
    wrapper = shallow(<AddCollection user={{uid: 1}} history={{push: jest.fn()}} />);
  })

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should handleChange', () => {
    wrapper.instance().handleChange(mockEvent)
    expect(wrapper.state('title')).toEqual('sweet')
  })


  it('should call postCollection on submit', () => {
    wrapper.find('form').simulate('submit', mockEvent);
    expect(postCollection).toBeCalled();
  })
})

