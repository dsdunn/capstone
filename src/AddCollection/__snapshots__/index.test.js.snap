// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCollection should match the snapShot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddCollection
    history={
        Object {
            "push": [Function],
          }
    }
    setCollection={[Function]}
    user={
        Object {
            "uid": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <a
          className="login-close-link"
          exact="true"
          onClick={[Function]}
>
          <img
                    className="close-btn"
                    src="close.svg"
          />
</a>,
        <label
          htmlFor="collection-image"
>
          Upload a picture of your collection
</label>,
        <input
          name="image"
          type="file"
/>,
        <label
          htmlFor="title"
>
          Collection name: 
</label>,
        <input
          name="title"
          onChange={[Function]}
          placeholder="my collection"
          value=""
/>,
        <label
          htmlFor="description"
>
          Describe your Collection:
</label>,
        <textarea
          name="description"
          onChange={[Function]}
          value=""
/>,
        <label
          htmlFor="collection-form-category"
>
          Select a Category
</label>,
        <div
          className="collection-form-bottom"
>
          <select
                    name="category"
                    onChange={[Function]}
                    required={true}
          >
                    <option
                              value=""
                    >
                              choose a category
                    </option>
                    <option
                              value="coins"
                    >
                              Coins
                    </option>
                    <option
                              value="comics"
                    >
                              Comics
                    </option>
                    <option
                              value="cards"
                    >
                              Cards
                    </option>
                    <option
                              value="vinyl"
                    >
                              Vinyl
                    </option>
                    <option
                              value="other"
                    >
                              Other
                    </option>
          </select>
          <button>
                    Submit
          </button>
</div>,
      ],
      "className": "add-collection-form",
      "onSubmit": [Function],
    },
    "ref": [Function],
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            className="close-btn"
            src="close.svg"
/>,
          "className": "login-close-link",
          "exact": "true",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "close-btn",
            "src": "close.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Upload a picture of your collection",
          "htmlFor": "collection-image",
        },
        "ref": null,
        "rendered": "Upload a picture of your collection",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "image",
          "type": "file",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Collection name: ",
          "htmlFor": "title",
        },
        "ref": null,
        "rendered": "Collection name: ",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "title",
          "onChange": [Function],
          "placeholder": "my collection",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Describe your Collection:",
          "htmlFor": "description",
        },
        "ref": null,
        "rendered": "Describe your Collection:",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "description",
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select a Category",
          "htmlFor": "collection-form-category",
        },
        "ref": null,
        "rendered": "Select a Category",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <select
              name="category"
              onChange={[Function]}
              required={true}
>
              <option
                            value=""
              >
                            choose a category
              </option>
              <option
                            value="coins"
              >
                            Coins
              </option>
              <option
                            value="comics"
              >
                            Comics
              </option>
              <option
                            value="cards"
              >
                            Cards
              </option>
              <option
                            value="vinyl"
              >
                            Vinyl
              </option>
              <option
                            value="other"
              >
                            Other
              </option>
</select>,
            <button>
              Submit
</button>,
          ],
          "className": "collection-form-bottom",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value=""
>
                  choose a category
</option>,
                <option
                  value="coins"
>
                  Coins
</option>,
                <option
                  value="comics"
>
                  Comics
</option>,
                <option
                  value="cards"
>
                  Cards
</option>,
                <option
                  value="vinyl"
>
                  Vinyl
</option>,
                <option
                  value="other"
>
                  Other
</option>,
              ],
              "name": "category",
              "onChange": [Function],
              "required": true,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "choose a category",
                  "value": "",
                },
                "ref": null,
                "rendered": "choose a category",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Coins",
                  "value": "coins",
                },
                "ref": null,
                "rendered": "Coins",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Comics",
                  "value": "comics",
                },
                "ref": null,
                "rendered": "Comics",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Cards",
                  "value": "cards",
                },
                "ref": null,
                "rendered": "Cards",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Vinyl",
                  "value": "vinyl",
                },
                "ref": null,
                "rendered": "Vinyl",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Other",
                  "value": "other",
                },
                "ref": null,
                "rendered": "Other",
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="login-close-link"
            exact="true"
            onClick={[Function]}
>
            <img
                        className="close-btn"
                        src="close.svg"
            />
</a>,
          <label
            htmlFor="collection-image"
>
            Upload a picture of your collection
</label>,
          <input
            name="image"
            type="file"
/>,
          <label
            htmlFor="title"
>
            Collection name: 
</label>,
          <input
            name="title"
            onChange={[Function]}
            placeholder="my collection"
            value=""
/>,
          <label
            htmlFor="description"
>
            Describe your Collection:
</label>,
          <textarea
            name="description"
            onChange={[Function]}
            value=""
/>,
          <label
            htmlFor="collection-form-category"
>
            Select a Category
</label>,
          <div
            className="collection-form-bottom"
>
            <select
                        name="category"
                        onChange={[Function]}
                        required={true}
            >
                        <option
                                    value=""
                        >
                                    choose a category
                        </option>
                        <option
                                    value="coins"
                        >
                                    Coins
                        </option>
                        <option
                                    value="comics"
                        >
                                    Comics
                        </option>
                        <option
                                    value="cards"
                        >
                                    Cards
                        </option>
                        <option
                                    value="vinyl"
                        >
                                    Vinyl
                        </option>
                        <option
                                    value="other"
                        >
                                    Other
                        </option>
            </select>
            <button>
                        Submit
            </button>
</div>,
        ],
        "className": "add-collection-form",
        "onSubmit": [Function],
      },
      "ref": [Function],
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              className="close-btn"
              src="close.svg"
/>,
            "className": "login-close-link",
            "exact": "true",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "close-btn",
              "src": "close.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Upload a picture of your collection",
            "htmlFor": "collection-image",
          },
          "ref": null,
          "rendered": "Upload a picture of your collection",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "image",
            "type": "file",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Collection name: ",
            "htmlFor": "title",
          },
          "ref": null,
          "rendered": "Collection name: ",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "title",
            "onChange": [Function],
            "placeholder": "my collection",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Describe your Collection:",
            "htmlFor": "description",
          },
          "ref": null,
          "rendered": "Describe your Collection:",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "description",
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select a Category",
            "htmlFor": "collection-form-category",
          },
          "ref": null,
          "rendered": "Select a Category",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <select
                name="category"
                onChange={[Function]}
                required={true}
>
                <option
                                value=""
                >
                                choose a category
                </option>
                <option
                                value="coins"
                >
                                Coins
                </option>
                <option
                                value="comics"
                >
                                Comics
                </option>
                <option
                                value="cards"
                >
                                Cards
                </option>
                <option
                                value="vinyl"
                >
                                Vinyl
                </option>
                <option
                                value="other"
                >
                                Other
                </option>
</select>,
              <button>
                Submit
</button>,
            ],
            "className": "collection-form-bottom",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value=""
>
                    choose a category
</option>,
                  <option
                    value="coins"
>
                    Coins
</option>,
                  <option
                    value="comics"
>
                    Comics
</option>,
                  <option
                    value="cards"
>
                    Cards
</option>,
                  <option
                    value="vinyl"
>
                    Vinyl
</option>,
                  <option
                    value="other"
>
                    Other
</option>,
                ],
                "name": "category",
                "onChange": [Function],
                "required": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "choose a category",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": "choose a category",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Coins",
                    "value": "coins",
                  },
                  "ref": null,
                  "rendered": "Coins",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Comics",
                    "value": "comics",
                  },
                  "ref": null,
                  "rendered": "Comics",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Cards",
                    "value": "cards",
                  },
                  "ref": null,
                  "rendered": "Cards",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Vinyl",
                    "value": "vinyl",
                  },
                  "ref": null,
                  "rendered": "Vinyl",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Other",
                    "value": "other",
                  },
                  "ref": null,
                  "rendered": "Other",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
